catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/security/AccessControlAccessPluginTest.java

```diff
@@ -13,154 +13,273 @@
 
-  private AccessPlugin accessPlugin;
-
-  @Mock private Attribute mockAttribute;
+  private static final String USER_EINSTEIN = "einstein@localhost.local";
 
-  @Mock private Metacard mockMetacard;
+  private static final String USER_DIJKSTRA = "dijkstra@localhost.local";
 
-  @Mock private Subject subject;
+  private static final String METACARD_ID = "100";
 
-  private SubjectIdentity subjectIdentity = mock(SubjectIdentity.class);
+  private AccessPlugin accessPlugin;
 
   @Before
   public void setUp() {
-    initMocks(this);
-    when(mockMetacard.getId()).thenReturn(MOCK_METACARD_ID);
-    when(subjectIdentity.getUniqueIdentifier(subject)).thenReturn(MOCK_IDENTITY_ATTR);
-    when(mockMetacard.getAttribute(Security.ACCESS_ADMINISTRATORS)).thenReturn(mockAttribute);
-    when(mockAttribute.getValues()).thenReturn(Collections.singletonList(MOCK_IDENTITY_ATTR));
+    setSubject(USER_EINSTEIN);
+  }
+
+  private void setSubject(String user) {
+    ThreadContext.unbindSubject();
+    Subject subject = mock(Subject.class);
+    SubjectIdentity subjectIdentity = mock(SubjectIdentity.class);
+    when(subjectIdentity.getUniqueIdentifier(subject)).thenReturn(user);
     ThreadContext.bind(subject);
 
     accessPlugin = new AccessControlAccessPlugin(subjectIdentity);
   }
 
-  private UpdateRequest mockUpdateRequest(Map<String, Metacard> updates) {
-    UpdateRequest update = mock(UpdateRequest.class);
-    doReturn(new ArrayList(updates.entrySet())).when(update).getUpdates();
-    return update;
+  private UpdateRequest getUpdateRequest(String id, Metacard metacard) {
+    return new UpdateRequestImpl(id, metacard);
+  }
+
+  @Test
+  public void testOwnerCanModify() throws StopProcessingException {
+    testUserWithAttributeCanModifyUnprotectedAttribute(Core.METACARD_OWNER);
+  }
+
+  @Test
+  public void testAccessIndividualsReadCanModify() throws StopProcessingException {
+    testUserWithAttributeCanModifyUnprotectedAttribute(Security.ACCESS_INDIVIDUALS_READ);
+  }
+
+  @Test
+  public void testAccessIndividualsCanModify() throws StopProcessingException {
+    testUserWithAttributeCanModifyUnprotectedAttribute(Security.ACCESS_INDIVIDUALS);
+  }
+
+  @Test
+  public void testAccessAdminsCanModify() throws StopProcessingException {
+    testUserWithAttributeCanModifyUnprotectedAttribute(Security.ACCESS_ADMINISTRATORS);
+  }
+
+  /**
+   * If an unprotected attribute was modified, then the plugin does not care who modified it and
+   * allows the request to continue. As of this writing, the {@code FilterPlugin} handles instances
+   * of a user with {@link Security#ACCESS_INDIVIDUALS_READ} attempting to modify any attributes on
+   * the metacard.
+   *
+   * <p>The choice of {@link Core#TITLE} for this set of tests was arbitrary. It is not an ACL
+   * attribute and therefore is sufficient for the needs of the test.
+   */
+  private void testUserWithAttributeCanModifyUnprotectedAttribute(String securityAttribute)
```

Comment:
:pencil2: Not JUnit tests should probably start with something other than test (for readability), for example validate/check.

Reply:
Good idea. I'll fix this.